### PR TITLE
Implementing MetAPI Cache

### PR DESCRIPTION
**Pull request by AnnaBanani**
Implementing MetAPI Cache

### CODE REVIEW

**Review by Ssimboss — CHANGES_REQUESTED** [review]

File: TheMET/TheMET/Network/MetAPICache.swift [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    func putObjectsResponce(metadataDate: Date?, departmentIds: [Int], responceData: ObjectsResponse) {
+        let newCachedData = CachedData<ObjectsRequestData, ObjectsResponse>(
+            requestData: ObjectsRequestData(metadataDate: metadataDate, departmentIds: departmentIds),
+            responceData: responceData,
+            date: Date.now
+        )
+        self.objectsCache.append(newCachedData)
+    }
+    
+    func putObjectResponce(requestData: ArtID, responceData: ObjectResponse) {
```
[the remark this answers is not in the document]
Comment (AnnaBanani) [reply TheMET/TheMET/Network/MetAPICache.swift:103]: +

File: TheMET/TheMET/Network/MetAPICache.swift [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    func putDepartmentsResponce(responceData: DepartmentsResponse) {
+        let newCachedData = CachedData<Void, DepartmentsResponse>(
+        requestData: Void(),
+        responceData: responceData,
+        date: Date.now
+        )
+        self.departmentsCashe.append(newCachedData)
+    }
+    
+    func putSearchResponce(requestData: [SearchParameter], responceData: SearchResponse) {
```
[the remark this answers is not in the document]
Comment (AnnaBanani) [reply TheMET/TheMET/Network/MetAPICache.swift:121]: +

File: TheMET/TheMET/Network/MetAPICache.swift [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            requestData: ObjectsRequestData(metadataDate: metadataDate, departmentIds: departmentIds),
+            responceData: responceData,
+            date: Date.now
+        )
+        self.objectsCache.append(newCachedData)
+    }
+    
+    func putObjectResponce(requestData: ArtID, responceData: ObjectResponse) {
+        let newCachedData = CachedData<ArtID, ObjectResponse>(
+        requestData: requestData,
```
[the remark this answers is not in the document]
Comment (AnnaBanani) [reply TheMET/TheMET/Network/MetAPICache.swift:105]: +

File: TheMET/TheMET/Network/MetAPICache.swift [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        responceData: responceData,
+        date: Date.now
+        )
+        self.objectCache.append(newCachedData)
+    }
+    
+    func putDepartmentsResponce(responceData: DepartmentsResponse) {
+        let newCachedData = CachedData<Void, DepartmentsResponse>(
+        requestData: Void(),
```
[the remark this answers is not in the document]
Comment (AnnaBanani) [reply TheMET/TheMET/Network/MetAPICache.swift:114]: +

File: TheMET/TheMET/Network/MetAPICache.swift [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        responceData: responceData,
+        date: Date.now
+        )
+        self.departmentsCashe.append(newCachedData)
+    }
+    
+    func putSearchResponce(requestData: [SearchParameter], responceData: SearchResponse) {
+        let newCachedData = CachedData<[SearchParameter], SearchResponse>(
+        requestData: requestData,
```
[the remark this answers is not in the document]
Comment (AnnaBanani) [reply TheMET/TheMET/Network/MetAPICache.swift:123]: +

File: TheMET/TheMET/Network/MetAPICache.swift [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+import Foundation
+
+class MetAPICashe {
+    
+    static let standard: MetAPICashe = MetAPICashe()
+    
+    private struct CachedData<RequestData, ResponceData> {
+        let requestData: RequestData
+        let responceData: ResponceData
+        let date: Date
```
[the remark this answers is not in the document]
Comment (AnnaBanani) [reply TheMET/TheMET/Network/MetAPICache.swift:17]: +

**Pull request merged**